Assets/CrowdControl/Scripts/TaskMoveToWayPoint.cs

```diff
@@ -0,0 +1,62 @@
+using System.Collections.Generic;
+using BehaviorTree;
+using NodeReflection;
+using UnityEngine;
+using UnityEngine.AI;
+
+namespace CrowdControl
+{
+	public class TaskMoveToWayPoint : Node
+	{
+		[ExposedInVisualEditor]
+		private float Radius { get; set; }
+		
+		//[ExposedInVisualEditor]
+		//private float DistanceToWayPoint { get; set; } //TODO use property when visual is implemented
+		private float DistanceToWayPoint = 5f;
```

Comment:
I think this node has too much variables considering its name, it should only move the agent to the waypoint. If you need a location check, implement another node, same thing for patience behavior.